src/db/artacle.js

```diff
@@ -1,3 +1,27 @@
+// Updates the artacle_projects table with the slug of the Artacle collection
+async function updateArtacleProjects({
+  client,
+  updates, // array of {projectId, slug} objects
+}) {
+  await client.query(
+    `
+    WITH updates AS (
+      SELECT
+        inputs.project_id,
+        inputs.artacle_slug
+      FROM unnest($1::projectid[], $2::text[]) as inputs(project_id, artacle_slug)
+    )
+    INSERT INTO artacle_projects (project_id, artacle_slug, update_time)
+    SELECT project_id, artacle_slug, now()
+    FROM updates
```

Comment:
This CTE works fine, but IMHO it's just as clear to use a simple
`unnest` in the `UPDATE ... FROM` clause, since we don't have any
auxiliary data to fetch based on the inputs? Thus:

```diff
   await client.query(
     `
-    WITH updates AS (
-      SELECT
-        inputs.project_id,
-        inputs.artacle_slug
-      FROM unnest($1::projectid[], $2::text[]) as inputs(project_id, artacle_slug)
-    )
     INSERT INTO artacle_projects (project_id, artacle_slug, update_time)
     SELECT project_id, artacle_slug, now()
-    FROM updates
+    FROM unnest($1::projectid[], $2::text[]) AS inputs(project_id, artacle_slug)
     ON CONFLICT (project_id) DO UPDATE
       SET artacle_slug = excluded.artacle_slug,
       update_time = excluded.update_time
```

Your tests still pass with this patch.